### PR TITLE
use SES region from envar

### PR DESCRIPTION
**Pull request by dp6ai**
[this](https://dev.chaskiq.io/en/articles/enviroment-variables-configuration) should probably be updated with the additional ENVAR requirement SES_REGION

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -118,6 +118,8 @@
     ActionMailer::Base.add_delivery_method :ses, AWS::SES::Base,
                                            access_key_id: ENV['AWS_ACCESS_KEY_ID'],
                                            secret_access_key: ENV['AWS_SECRET_ACCESS_KEY'],
+                                           server: "email.#{ENV['SES_REGION']}.amazonaws.com",
+                                           message_id_domain: "#{ENV['SES_REGION']}.amazonses.com",
```
Comment (michelson) [inline config/environments/production.rb]: we should have this as an optional parameter, maybe use fetch ENV.fetch() to give the defaults to us-east-1 , this way we will not break existing installations

**Comment by dp6ai** [comment]
Good point, hopefully this is pretty explicit

**Comment by dp6ai** [comment]
Still getting an error message on this `SignatureDoesNotMatch - Credential should be scoped to a valid region, not 'us-east-1'.` I'll investigate more

**Comment by dp6ai** [comment]
@michelson i feel like this is how i ended up using the other gem last time. 
with ses_config `signature_version=>2`
```
{:access_key_id=>"XXXXXXXX", :secret_access_key=>"YYYYYYYYY", :signature_version=>2, :server=>"email.eu-west-2.amazonaws.com", :message_id_domain=>"eu-west-2.amazonses.com"}
```

i get `InvalidParameterValue - Invalid date 20210622T222400Z. It must be in one of the formats specified by HTTP RFC 2616 section 3.3.1`

and with `signature_version=>4`

```
{:access_key_id=>"XXXXXXXX", :secret_access_key=>"YYYYYYYYY", :signature_version=>4, :server=>"email.eu-west-2.amazonaws.com", :message_id_domain=>"eu-west-2.amazonses.com"}
```

i get `SignatureDoesNotMatch - Credential should be scoped to a valid region, not 'us-east-1'.`


Pretty frustrating but i'll take another look at it tomorrow. If you have any thoughts, let me know

**Comment by michelson** [comment]
I've seen this when the ses account is created on a zone but the IAM is from another zone

**Comment by dp6ai** [comment]
But weirdly my confit works fine with mail-ses gem. Can you send me exactly the config layout you are using with non standard regions and I'll test. 

Thanks

**Comment by dp6ai** [comment]
Just to add to this, my AWS console has this message under IAM's `IAM does not require region selection.`

**Comment by dp6ai** [comment]
@michelson I'm going to see if i can change to the the official [gem](https://github.com/aws/aws-sdk-rails) as per [here](https://github.com/drewblas/aws-ses/issues/78) (without breaking existing users)
You can choose whether to accept

**Comment by dp6ai** [comment]
I'll resubmit this when i have it working

**Pull request closed without merging**